### PR TITLE
Remove need to double deserialize EnvelopeHeader to verify msg signature

### PR DESCRIPTION
**Pull request by sdbondi**
## Description

Moved verify signature into MessageEnvelopeHeader so that
MessageEnvelope doesn't have to deserialize the header to verify.

## Motivation and Context

Small optimisation of message signature verification

## How Has This Been Tested?

Split up a few unit tests and added a new one specifically for signature verification 

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: comms/src/message/envelope.rs [hunk 1]
```diff
@@ -130,20 +121,22 @@ impl MessageEnvelope {
...
 
     /// Returns the Message deserialized from the body frame
-    pub fn message_body(&self) -> Result<Message, MessageError> {
+    pub fn to_message(&self) -> Result<Message, MessageError> {
         Message::from_binary(self.body_frame()).map_err(Into::into)
     }
 
     /// Returns the decrypted and deserialized Message from the body frame
-    pub fn decrypted_message_body<PK>(
+    pub fn to_decrypted_message<PK>(
```
Comment (SWvheerden) [inline comms/src/message/envelope.rs]: Nit: We have you renamed this? To_XX makes me think you are converting the body, which you are not doing.

File: comms/src/message/envelope.rs [hunk 2]
```diff
@@ -130,20 +121,22 @@ impl MessageEnvelope {
     }
 
     /// Returns the Message deserialized from the body frame
-    pub fn message_body(&self) -> Result<Message, MessageError> {
+    pub fn to_message(&self) -> Result<Message, MessageError> {
```
Comment (SWvheerden) [inline comms/src/message/envelope.rs]: Nit: We have you renamed this? To_XX makes me think you are converting the body, which you are not doing.

Comment (sdbondi) [reply comms/src/message/envelope.rs]: That's exactly what i'm doing ;) (deserializing the body)

File: comms/src/message/envelope.rs (hunk 1 above)
Comment (sdbondi) [reply comms/src/message/envelope.rs]: That's exactly what i'm doing ;) (decrypting and deserializing the body)

File: comms/src/message/envelope.rs (hunk 2 above)
Comment (sdbondi) [reply comms/src/message/envelope.rs]: I've changed the method names to `deserialize_xxx` (header, message, encrypted_message)

**Pull request merged**